src/api/users/src/routes/users.js

```diff
@@ -28,29 +28,50 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
 router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+  const page = !parseInt(req.query.page, 10) ? 1 : parseInt(req.query.page, 10); // if req.query.page is 0 or unparsable set to 1, otherwise set to the req.query.page
+  const perPage = !parseInt(req.query.per_page, 10) ? 5 : parseInt(req.query.per_page, 10);
+  const usersArray = [];
+  let userToStartAt;
 
-    if (snapshot.empty) {
-      logger.debug(`All users were requested by ${req.ip} but could not be found.`);
-      res.status(404).json({
-        msg: `That collection could not be found, or contains no data.`,
-      });
+  try {
+    if (page === 1) {
+      userToStartAt = 0;
+    } else if (page === 0) {
+      userToStartAt = perPage;
     } else {
-      logger.debug(`All users were requested by ${req.ip} and served successfully.`);
+      userToStartAt = perPage * (page - 1);
+    }
 
-      const usersArray = [];
-      snapshot.forEach((doc) => {
+    const query = await db
+      .collection('users')
+      .orderBy('id')
+      .startAt(userToStartAt)
+      .limit(perPage)
+      .get();
+
+    // if we're requesting too many users, return the last n (perPage) users in the db
```

Comment:
What does this mean?

Reply:
Say a user requests something like `/?per_page=5&page=100` but only 1 user exists in the db, Firebase returns an empty array, hence `if (!query.size)`. I set it up to return the last `perPage` amount of users in this case.

I'm failing at succinctly saying that here.

Reply:
If I request a page that doesn't exist, give back the empty array.  I think Firebase's behaviour is correct, and we should do the same.

Reply:
kk will remove then